src/platform_impl/linux/wayland/event_loop/mod.rs

```diff
@@ -34,6 +35,9 @@ pub use state::WinitState;
 
 use sink::EventSink;
 
+type WinitDispatcher =
+    calloop::Dispatcher<WaylandSource, fn((), &mut EventQueue, &mut WinitState) -> IOResult<u32>>;
```

Comment:
The second type parameter of `calloop::Dispatcher` is supposed to be the `Data` type of the calloop event loop, not the type of the closure (which is internally boxed). So tbh I'm surprised this type event works?

Reply:
I'm not sure how to use it then. Assuming that I'd use `()` for `Data` I'm getting.

```
   --> src/platform_impl/linux/wayland/event_loop/mod.rs:192:33
    |
134 |           let wayland_dispatcher = calloop::Dispatcher::new(wayland_source, |_, queue, ddata| {
    |  ___________________________________________________________________________-
135 | |             queue.dispatch_pending(ddata, |event, object, _| {
136 | |                 panic!(
137 | |                     "[calloop] Encountered an orphan event: {}@{} : {}",
...   |
142 | |             })
143 | |         });
    | |_________- the found closure
...
192 |               wayland_dispatcher: wayland_dispatcher.clone(),
    |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found closure
    |
    = note: expected struct `RefCell<Dispatcher<_, ()>>`
               found struct `RefCell<Dispatcher<_, [closure@src/platform_impl/linux/wayland/event_loop/mod.rs:134:75: 143:10]>>`

```
So it's saying that the type I'm providing when building dispatcher is closure and the result is Dispatcher<WaylandSource, closure>.

If you have an example how to use it it'll be nice.

Reply:
Oh, right, my bad. I didn't realize that SCTK still depended on calloop 0.7... This part of calloop's API has been changed in 0.8...